pkg/splunk/enterprise/util.go

```diff
@@ -69,6 +69,86 @@ func getIndexerExtraEnv(cr splcommon.MetaObject, replicas int32) []corev1.EnvVar
 	}
 }
 
+// getClusterMasterExtraEnv returns extra environment variables used by indexer clusters
+func getClusterMasterExtraEnv(cr splcommon.MetaObject, spec *enterprisev1.CommonSplunkSpec) []corev1.EnvVar {
+	if spec.ClusterMasterRef.Name != "" {
+		clusterMasterURL := GetSplunkServiceName(SplunkClusterMaster, spec.ClusterMasterRef.Name, false)
+		if spec.ClusterMasterRef.Namespace != "" {
+			clusterMasterURL = splcommon.GetServiceFQDN(spec.ClusterMasterRef.Namespace, clusterMasterURL)
+		}
+		return []corev1.EnvVar{
+			{
+				Name:  "SPLUNK_CLUSTER_MASTER_URL",
+				Value: clusterMasterURL,
+			},
+		}
+	}
+	return []corev1.EnvVar{
+		{
+			Name:  "SPLUNK_CLUSTER_MASTER_URL",
+			Value: GetSplunkServiceName(SplunkClusterMaster, cr.GetName(), false),
+		},
+	}
+}
+
+// getStandaloneExtraEnv returns extra environment variables used by monitoring console
+func getStandaloneExtraEnv(cr splcommon.MetaObject, replicas int32) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  "SPLUNK_STANDALONE_URL",
+			Value: GetSplunkStatefulsetUrls(cr.GetNamespace(), SplunkStandalone, cr.GetName(), replicas, false),
+		},
+	}
+}
+
+// getLicenseMasterURL returns URL of license master
+func getLicenseMasterURL(cr splcommon.MetaObject, spec *enterprisev1.CommonSplunkSpec) []corev1.EnvVar {
+	if spec.LicenseMasterRef.Name != "" {
+		licenseMasterURL := GetSplunkServiceName(SplunkLicenseMaster, spec.LicenseMasterRef.Name, false)
+		if spec.LicenseMasterRef.Namespace != "" {
+			licenseMasterURL = splcommon.GetServiceFQDN(spec.LicenseMasterRef.Namespace, licenseMasterURL)
+		}
```

Comment:
Same as the case of Licence Master. I may be missing something here, but please let me know if we don't need FQDN in a certain cases?
Wondering how this is different from a case when we are dealing with the REST calls..

Reply:
For rest calls we are using fqdn service, For license master, we are not using it anywhere. This is just for MC to add LM as its peer